client/alloc_runner.go

```diff
@@ -926,6 +933,9 @@ func (r *AllocRunner) handleDestroy() {
 	// state as we wait for a destroy.
 	alloc := r.Alloc()
 
+	// Incr the alloc destroy counter
```

Comment:
Like above, I don't think we ever really document or expose what "destroy" means, so this metric seems confusing to end users. Perhaps expand the comment with something like:

```go
// Increment the destroy count for this alloc runner since this allocation is being removed from this client.
```